packages/token-bridge-sdk/src/hooks/useArbTokenBridge.ts

```diff
@@ -137,6 +175,38 @@ export const useArbTokenBridge = (
 
   const [l11NetworkID, setL1NetWorkID] = useState<string | null>(null)
 
+  async function getL1TokenData(erc20L1Address: string) {
+    if (typeof l1.signer.provider === 'undefined') {
+      throw new Error(`No instance of L1Provider found`)
+    }
+
+    const erc20Bridger = new Erc20Bridger(l2.network)
+    const l1GatewayAddress = await erc20Bridger.getL1GatewayAddress(
+      erc20L1Address,
+      l1.signer.provider
+    )
+
+    const contract = ERC20__factory.connect(erc20L1Address, l1.signer)
+
+    const multiCaller = await MultiCaller.fromProvider(l1.signer.provider)
+    const [tokenData] = await multiCaller.getTokenData([erc20L1Address], {
+      name: true,
+      symbol: true,
+      balanceOf: { account: walletAddress },
+      allowance: { owner: walletAddress, spender: l1GatewayAddress },
+      decimals: true
+    })
+
+    return {
+      name: tokenData.name || getDefaultTokenName(erc20L1Address),
+      symbol: tokenData.symbol || getDefaultTokenSymbol(erc20L1Address),
+      balance: tokenData.balance || BigNumber.from(0),
+      allowance: tokenData.allowance || BigNumber.from(0),
+      decimals: tokenData.decimals || 0,
```

Comment:
Added defaults for optional ERC-20 methods: `name`, `symbol`, and `decimals`. 
Also added defaults for `balance` and `allowance`, but unsure if needed?

Reply:
R.e. falling back to defaults, we should just be mindful of the fact that (I believe) this function won't revert if given a non-token contract address; fine as is since it's only called in `depositToken`, but might be worth leaving a comment